File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    return Helper(root, 0)
};

function Helper(root, totalSum){
    if(root === null) return 0;
    totalSum = totalSum*10 + root.val;
    if(root.left === null && root.right === null ) return totalSum;
    return Helper(root.left, totalSum) + Helper(root.right, totalSum);
}

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.Helper(root, 0)
         

    def Helper(self, node, totalSum):
        if node == None:
            return 0
        totalSum = totalSum*10 + node.val
        if node.left == None and node.right == None:
            return totalSum
        return self.Helper(node.left, totalSum) + self.Helper(node.right, totalSum)
        
        